src/collect_objects.py

```diff
@@ -0,0 +1,55 @@
+import dataclasses
+import json
+import os
+from typing import List, Type
+
+from dacite import Config as DaciteConfig
+from dacite import from_dict
+
+from db_objects import Query
+
+@dataclasses.dataclass
+class CollectResult:
+    db_version: str = ""
+    git_message: str = ""
+    config: str = ""
+    model_queries: List[str] = None
+    queries: List[Type[Query]] = None
+
+    def append(self, new_element):
+        if not self.queries:
+            self.queries = [new_element, ]
+        else:
+            self.queries.append(new_element)
+
+        # CPUs are cheap
+        self.queries.sort(key=lambda q: q.query_hash)
+
+    def find_query_by_hash(self, query_hash) -> Type[Query] | None:
+        return next(
```

Comment:
I think [bisect.bisect_left](https://docs.python.org/3/library/bisect.html) would be more intuitive, and should be also faster except for very small models.